lib/minheap.js

```diff
@@ -11,18 +11,36 @@ class MinHeap {
   }
 
   // This method adds a HeapNode instance to the heap
-  // Time Complexity: ?
-  // Space Complexity: ?
+  // Time Complexity: O(log n)
+  // Space Complexity: O(1)
   add(key, value = key) {
-    throw new Error("Method not implemented yet...");
+    // create new HeapNode
+    const newNode = new HeapNode(key, value);
+
+    // add the node to the end
+    this.store.push(newNode);
+
+    // perform heap-up where parent must be less than children
+    const index = this.store.length - 1;
+    this.heapUp(index);
   }
 
   // This method removes and returns an element from the heap
   //   maintaining the heap structure
-  // Time Complexity: ?
-  // Space Complexity: ?
+  // Time Complexity: O(log n)
+  // Space Complexity: O(1)
   remove() {
```

Comment:
✨ However space complexity will be O(log n) here because of the recursive call stack of `heap_down`